vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinDefaultRequestCache.java

```diff
@@ -0,0 +1,69 @@
+package com.vaadin.flow.spring.security;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.vaadin.flow.server.connect.EndpointUtil;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.security.web.savedrequest.HttpSessionRequestCache;
+import org.springframework.stereotype.Component;
+
+/**
+ * A default request cache implementation which aims to ignore requests that are
+ * not for routes.
+ * 
+ * Using this class helps with redirecting the user to the correct route after
+ * login instead of redirecting to some internal URL like a service worker or
+ * some data the service worker has fetched.
+ */
+@Component
+public class VaadinDefaultRequestCache extends HttpSessionRequestCache {
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Autowired
+    private RequestUtil requestUtil;
+
+    private Object endpointUtil;
```

Comment:
Don't need to use Object, it can actually be `EndpointUtil endpointUtil`

Reply:
How? It results in `Caused by: java.lang.IllegalStateException: Failed to introspect Class [com.vaadin.flow.spring.security.VaadinDefaultRequestCache] from ClassLoader [sun.misc.Launcher$AppClassLoader@18b4aac2]`